galsim/utilities.py

```diff
@@ -330,34 +349,137 @@ def thin_tabulated_values(x, f, rel_err=1.e-4, preserve_range=False):
         # Nothing to do
         return x,f
 
+    total_integ = np.trapz(abs(f), x)
+    thresh = total_integ * rel_err
+
+    if not preserve_range:
+        errleft = 0.5 * (abs(f[1])+abs(f[0])) * (x[1]-x[0])
+        errright = 0.5 * (abs(f[-1])+abs(f[-2])) * (x[-1]-x[-2])
+        accumulated_err = 0.0
+        # arbitrarily allocate half of error budget to trimming the endpoints.
+        # while at least one endpoint can be trimmed:
```

Comment:
Good idea.  I just copied the code from old_thin_tabulated_values().  Here's the new test_thin() dump:

```
Original number of SED samples =  280
Test err =  0.01
num samples with preserve_range = True:  55
realized error =  -0.000283460094106
num samples with preserve_range = False:  58
realized error =  0.000853972319082
Test err =  0.001
num samples with preserve_range = True:  198
realized error =  0.000103143610595
num samples with preserve_range = False:  198
realized error =  0.000103143610595
Test err =  0.0001
num samples with preserve_range = True:  251
realized error =  -6.4320318584e-07
num samples with preserve_range = False:  251
realized error =  -6.4320318584e-07
Test err =  1e-05
num samples with preserve_range = True:  270
realized error =  6.32245433271e-06
num samples with preserve_range = False:  270
realized error =  6.32245433271e-06
```

Interestingly, it frequently chooses to not trim the edges at all.  Definitely an improvement in total number of retained samples too.  I'll rerun Jim's script and report back.